cupla.spec

```diff
@@ -0,0 +1,48 @@
+### RPM external cupla 0.2.0
+
+Source: https://github.com/ComputationalRadiationPhysics/%{n}/archive/%{realversion}.tar.gz
+Requires: alpaka
+Requires: cuda
+Requires: tbb
+
+%prep
+%setup -n %{n}-%{realversion}
+
+%build
+mkdir build lib
+
+# remove the version of Alpaka bundled with Cupla
+rm -rf alpaka
+
+CXXFLAGS="-m64 -std=c++14 -g -O3 -DALPAKA_DEBUG=0 -I$CUDA_ROOT/include -I$TBB_ROOT/include -I$BOOST_ROOT/include -I$ALPAKA_ROOT/include -Iinclude"
+HOST_FLAGS="-pthread -fPIC -Wall -Wextra -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-local-typedefs -Wno-attributes -Wno-reorder -Wno-sign-compare"
+NVCC_FLAGS="-w --generate-line-info --source-in-ptx --cudart=shared --expt-extended-lambda --expt-relaxed-constexpr -gencode arch=compute_35,code=sm_35 -gencode arch=compute_60,code=sm_60 -gencode arch=compute_70,code=sm_70"
```

Comment:
These flags should match those in [cuda-toolfile.spec](https://github.com/cms-sw/cmsdist/blob/IB/CMSSW_11_1_X/master/cuda-toolfile.spec#L37-L48).
@smuzaffar is there any way to make it automatic ?

Reply:
@fwyzard , only automatic way is to put these flags in a separate file e.g. `cuda-flags.file` and then include that file in both cuda-toolfile and here
```
## INLCUDE cuda-flags
```

Reply:
Mhm... I see.
However, the way in which the flags are used is a bit different:
in `cupla.spec` (simplified):
```bash
NVCC_FLAGS="--generate-line-info --source-in-ptx --expt-extended-lambda --expt-relaxed-constexpr -gencode arch=compute_35,code=sm_35 -gencode arch=compute_60,code=sm_60 -gencode arch=compute_70,code=sm_70 --cudart=shared"
```
vs `cuda-toolfile.spec` (simplified):
```bash
  <flags CUDA_FLAGS="--generate-line-info --source-in-ptx --expt-relaxed-constexpr --expt-extended-lambda -gencode arch=compute_35,code=sm_35 -gencode arch=compute_60,code=sm_60 -gencode arch=compute_70,code=sm_70 --cudart=shared "/>
```

Would it work to define a spec macro in a separate file, e.g. `cuda-flags.file`
```spec
%define cuda_flags -O3 -std=c++14 --expt-relaxed-constexpr --expt-extended-lambda
%define cuda_flags %cuda_flags --generate-line-info --source-in-ptx
%ifarch x86_64 ppc64le
%define cuda_flags %cuda_flags -gencode arch=compute_35,code=sm_35
%define cuda_flags %cuda_flags -gencode arch=compute_60,code=sm_60
%define cuda_flags %cuda_flags -gencode arch=compute_70,code=sm_70
%endif
%ifarch aarch64
%define cuda_flags %cuda_flags -gencode arch=compute_70,code=sm_70
%define cuda_flags %cuda_flags -gencode arch=compute_72,code=sm_72
%endif
%define cuda_flags %cuda_flags --cudart=shared
```
and then include it with something like
```spec
## IMPORT cuda-flags
...
NVCC_FLAGS="%{cuda-flags}"
```
and
```spec
## IMPORT cuda-flags
...
  <flags CUDA_FLAGS="%{cuda-flags}"/>
```
?

Reply:
yes macro will work